x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -427,5 +437,21 @@ private static Tuple<Mode, String> parsePolicyName(Token policyToken) {
         return new Tuple<>(mode, policyName);
     }
 
+    @Override
+    public LogicalPlan visitMetricsCommand(EsqlBaseParser.MetricsCommandContext ctx) {
+        if (Build.current().isSnapshot() == false) {
+            throw new IllegalArgumentException("METRICS command currently requires a snapshot build");
+        }
+        Source source = source(ctx);
+        TableIdentifier table = new TableIdentifier(source, null, visitMetricsIdentifiers(ctx.metricsIdentifier()));
+        final Stats stats;
+        if (ctx.aggregates != null || ctx.grouping != null) {
+            stats = stats(source, ctx.grouping, ctx.aggregates);
+        } else {
+            stats = new Stats(List.of(), List.of());
+        }
+        return new UnresolvedMetrics(source, table, null, stats.groupings, stats.aggregates());
```

Comment:
Based on the resolution rule, UnresolvedMetrics doesn't carry any extra information - instead of introducing this temporary object (which doesn't survive the analysis phase) simply return the aggregate plus unresolved relation (which will get resolved automatically):

```
visitMetricsCommand(...) {
  ...
val relation = new EsqlUnresolvedRelation(source, index, ...);
return input -> new EsqlAggregate(source(ctx), relation, stats.grouping, stats.aggregates);
} 
```

Reply:
We need to introduce this relation for the METRICS command so that we can carry the filter and map it to the TimeSeriesSortedSourceOperatorFactory. However, it's better to remove it now from this PR to avoid confusion. I've removed it in https://github.com/elastic/elasticsearch/pull/108115/commits/435ab0082cccca3a28f3178d829dab4c91564201.